crates/librqbit/README.md

```diff
@@ -1 +1,128 @@
+
+```rust
+use std::error::Error;
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::time::Duration;
+use librqbit::{http_api_client, Magnet};
+use librqbit::session::{AddTorrentResponse, ListOnlyResponse, ManagedTorrentState, Session};
+use librqbit::spawn_utils::BlockingSpawner;
+use size_format::SizeFormatterBinary as SF;
+use tokio::spawn;
+
+const MAGNET_LINK: &str = "magnet:?..."; // Put your magnet link here
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>>{
+
+    let spawner = BlockingSpawner::new(false);
+
+    // This function will print the torrent properties every 1s
+    let stats_printer = |session: Arc<Session>| async move {
+        loop {
+            session.with_torrents(|torrents| {
+                for (idx, torrent) in torrents.iter().enumerate() {
+                    match &torrent.state {
+                        ManagedTorrentState::Initializing => {
+                            println!("[{}] initializing", idx);
+                        },
+                        ManagedTorrentState::Running(handle) => {
+                            let peer_stats = handle.torrent_state().peer_stats_snapshot();
+                            let stats = handle.torrent_state().stats_snapshot();
+                            let speed = handle.speed_estimator();
+                            let total = stats.total_bytes;
+                            let progress = stats.total_bytes - stats.remaining_bytes;
+                            let downloaded_pct = if stats.remaining_bytes == 0 {
+                                100f64
+                            } else {
+                                (progress as f64 / total as f64) * 100f64
+                            };
+                            println!(
+                                "[{}]: {:.2}% ({:.2}), down speed {:.2} MiB/s, fetched {}, remaining {:.2} of {:.2}, uploaded {:.2}, peers: {{live: {}, connecting: {}, queued: {}, seen: {}}}",
+                                idx,
+                                downloaded_pct,
+                                SF::new(progress),
+                                speed.download_mbps(),
+                                SF::new(stats.fetched_bytes),
+                                SF::new(stats.remaining_bytes),
+                                SF::new(total),
+                                SF::new(stats.uploaded_bytes),
+                                peer_stats.live,
+                                peer_stats.connecting,
+                                peer_stats.queued,
+                                peer_stats.seen,
+                            );
+                        },
+                    }
+                }
+            });
+            tokio::time::sleep(Duration::from_secs(1)).await;
+        }
+    };
+
+    // Create the torrent session, one session can have more than one torrent at the same time
+    let session = Arc::new(
+        Session::new(PathBuf::from("C:\\Anime"), spawner).await?

```

Comment:
Maybe we can modify Session::new to take ```impl AsRef<Path>``` as argument, or at least use ```"C:\\Anime".into()``` for brevity